### PR TITLE
Add f90wrap

### PR DESCRIPTION
**Pull request by ocefpaf**
@jameskermode would you be interested in becoming a maintainer for this conda recipe?

### CODE REVIEW

**Comment by ocefpaf** [comment]
I have no idea how to fix the OS X error. I am using the default channel gcc to provide a Fortran compiler.

```shell
error: Command "/Users/travis/miniconda/envs/_build/bin/gfortran -Wall -g -arch x86_64 build/temp.macosx-10.5-x86_64-2.7/build/src.macosx-10.5-x86_64-2.7/f90wrap/sizeof_fortran_tmodule.o build/temp.macosx-10.5-x86_64-2.7/build/src.macosx-10.5-x86_64-2.7/fortranobject.o build/temp.macosx-10.5-x86_64-2.7/f90wrap/sizeoffortran.o -L/Users/travis/miniconda/envs/_build/lib/gcc/x86_64-apple-darwin11.4.2/4.8.5 -L/Users/travis/miniconda/envs/_build/lib -lgfortran -o build/lib.macosx-10.5-x86_64-2.7/f90wrap/sizeof_fortran_t.so" failed with exit status 1
```

(See https://travis-ci.org/conda-forge/staged-recipes/jobs/125708635)

**Comment by pelson** [comment]
There is more to the error:

```
Undefined symbols for architecture x86_64:

      "_PyArg_ParseTupleAndKeywords", referenced from:

          _f2py_rout_sizeof_fortran_t_sizeof_fortran_t in sizeof_fortran_tmodule.o

      "_PyCObject_AsVoidPtr", referenced from:

          _initsizeof_fortran_t in sizeof_fortran_tmodule.o

          _F2PyCapsule_AsVoidPtr in fortranobject.o

      "_PyCObject_FromVoidPtr", referenced from:

          _fortran_getattr in fortranobject.o

          _F2PyCapsule_FromVoidPtr in fortranobject.o

      "_PyCObject_Type", referenced from:

          _initsizeof_fortran_t in sizeof_fortran_tmodule.o

          _F2PyCapsule_Check in fortranobject.o

      "_PyDict_DelItemString", referenced from:

          _fortran_setattr in fortranobject.o

      "_PyDict_GetItemString", referenced from:

          _fortran_getattr in fortranobject.o

      "_PyDict_New", referenced from:

          _PyFortranObject_New in fortranobject.o

          _PyFortranObject_NewAsAttr in fortranobject.o

          _fortran_setattr in fortranobject.o

      "_PyDict_SetItemString", referenced from:

          _initsizeof_fortran_t in sizeof_fortran_tmodule.o

          _fortran_getattr in fortranobject.o

          _F2PyDict_SetItemString in fortranobject.o

          _PyFortranObject_New in fortranobject.o

          _fortran_setattr in fortranobject.o

      "_PyErr_Clear", referenced from:

          _fortran_repr in fortranobject.o

          _F2PyDict_SetItemString in fortranobject.o

      "_PyErr_Format", referenced from:

          _initsizeof_fortran_t in sizeof_fortran_tmodule.o

          _fortran_call in fortranobject.o

      "_PyErr_NewException", referenced from:

          _initsizeof_fortran_t in sizeof_fortran_tmodule.o

      "_PyErr_NoMemory", referenced from:

          _fortran_getattr in fortranobject.o

      "_PyErr_Occurred", referenced from:

          _f2py_rout_sizeof_fortran_t_sizeof_fortran_t in sizeof_fortran_tmodule.o

          _initsizeof_fortran_t in sizeof_fortran_tmodule.o

          _F2PyDict_SetItemString in fortranobject.o

      "_PyErr_Print", referenced from:

          _initsizeof_fortran_t in sizeof_fortran_tmodule.o

          _F2PyDict_SetItemString in fortranobject.o

      "_PyErr_SetString", referenced from:

          _initsizeof_fortran_t in sizeof_fortran_tmodule.o

          _array_from_pyobj in fortranobject.o

          _fortran_setattr in fortranobject.o

      "_PyExc_AttributeError", referenced from:

          _initsizeof_fortran_t in sizeof_fortran_tmodule.o

          _fortran_setattr in fortranobject.o

      "_PyExc_ImportError", referenced from:

          _initsizeof_fortran_t in sizeof_fortran_tmodule.o

      "_PyExc_RuntimeError", referenced from:

          _initsizeof_fortran_t in sizeof_fortran_tmodule.o

          _fortran_call in fortranobject.o

      "_PyExc_TypeError", referenced from:

          _fortran_call in fortranobject.o

          _array_from_pyobj in fortranobject.o

      "_PyExc_ValueError", referenced from:

          _array_from_pyobj in fortranobject.o

      "_PyImport_ImportModule", referenced from:

          _initsizeof_fortran_t in sizeof_fortran_tmodule.o

      "_PyMem_Free", referenced from:

          _fortran_dealloc in fortranobject.o

          _fortran_getattr in fortranobject.o

      "_PyMem_Malloc", referenced from:

          _fortran_getattr in fortranobject.o

      "_PyModule_GetDict", referenced from:

          _initsizeof_fortran_t in sizeof_fortran_tmodule.o

      "_PyOS_snprintf", referenced from:

          _fortran_getattr in fortranobject.o

      "_PyObject_GetAttrString", referenced from:

          _initsizeof_fortran_t in sizeof_fortran_tmodule.o

          _fortran_repr in fortranobject.o

      "_PyString_AsString", referenced from:

          _fortran_repr in fortranobject.o

      "_PyString_ConcatAndDel", referenced from:

          _fortran_getattr in fortranobject.o

      "_PyString_FromFormat", referenced from:

          _fortran_repr in fortranobject.o

      "_PyString_FromString", referenced from:

          _initsizeof_fortran_t in sizeof_fortran_tmodule.o

          _fortran_repr in fortranobject.o

          _fortran_getattr in fortranobject.o

      "_PyString_FromStringAndSize", referenced from:

          _fortran_getattr in fortranobject.o

      "_PyType_IsSubtype", referenced from:

          _array_from_pyobj in fortranobject.o

      "_PyType_Type", referenced from:

          _initsizeof_fortran_t in sizeof_fortran_tmodule.o

      "_Py_BuildValue", referenced from:

          _f2py_rout_sizeof_fortran_t_sizeof_fortran_t in sizeof_fortran_tmodule.o

      "_Py_FindMethod", referenced from:

          _fortran_getattr in fortranobject.o

      "_Py_InitModule4_64", referenced from:

          _initsizeof_fortran_t in sizeof_fortran_tmodule.o

      "__PyObject_New", referenced from:

          _PyFortranObject_New in fortranobject.o

          _PyFortranObject_NewAsAttr in fortranobject.o

      "__Py_NoneStruct", referenced from:

          _fortran_getattr in fortranobject.o

          _array_from_pyobj in fortranobject.o

          _fortran_setattr in fortranobject.o

      "_main", referenced from:

          start in crt1.10.6.o

    ld: symbol(s) not found for architecture x86_64

    collect2: error: ld returned 1 exit status

    Undefined symbols for architecture x86_64:

      "_PyArg_ParseTupleAndKeywords", referenced from:

          _f2py_rout_sizeof_fortran_t_sizeof_fortran_t in sizeof_fortran_tmodule.o

      "_PyCObject_AsVoidPtr", referenced from:

          _initsizeof_fortran_t in sizeof_fortran_tmodule.o

          _F2PyCapsule_AsVoidPtr in fortranobject.o

      "_PyCObject_FromVoidPtr", referenced from:

          _fortran_getattr in fortranobject.o

          _F2PyCapsule_FromVoidPtr in fortranobject.o

      "_PyCObject_Type", referenced from:

          _initsizeof_fortran_t in sizeof_fortran_tmodule.o

          _F2PyCapsule_Check in fortranobject.o

      "_PyDict_DelItemString", referenced from:

          _fortran_setattr in fortranobject.o

      "_PyDict_GetItemString", referenced from:

          _fortran_getattr in fortranobject.o

      "_PyDict_New", referenced from:

          _PyFortranObject_New in fortranobject.o

          _PyFortranObject_NewAsAttr in fortranobject.o

          _fortran_setattr in fortranobject.o

      "_PyDict_SetItemString", referenced from:

          _initsizeof_fortran_t in sizeof_fortran_tmodule.o

          _fortran_getattr in fortranobject.o

          _F2PyDict_SetItemString in fortranobject.o

          _PyFortranObject_New in fortranobject.o

          _fortran_setattr in fortranobject.o

      "_PyErr_Clear", referenced from:

          _fortran_repr in fortranobject.o

          _F2PyDict_SetItemString in fortranobject.o

      "_PyErr_Format", referenced from:

          _initsizeof_fortran_t in sizeof_fortran_tmodule.o

          _fortran_call in fortranobject.o

      "_PyErr_NewException", referenced from:

          _initsizeof_fortran_t in sizeof_fortran_tmodule.o

      "_PyErr_NoMemory", referenced from:

          _fortran_getattr in fortranobject.o

      "_PyErr_Occurred", referenced from:

          _f2py_rout_sizeof_fortran_t_sizeof_fortran_t in sizeof_fortran_tmodule.o

          _initsizeof_fortran_t in sizeof_fortran_tmodule.o

          _F2PyDict_SetItemString in fortranobject.o

      "_PyErr_Print", referenced from:

          _initsizeof_fortran_t in sizeof_fortran_tmodule.o

          _F2PyDict_SetItemString in fortranobject.o

      "_PyErr_SetString", referenced from:

          _initsizeof_fortran_t in sizeof_fortran_tmodule.o

          _array_from_pyobj in fortranobject.o

          _fortran_setattr in fortranobject.o

      "_PyExc_AttributeError", referenced from:

          _initsizeof_fortran_t in sizeof_fortran_tmodule.o

          _fortran_setattr in fortranobject.o

      "_PyExc_ImportError", referenced from:

          _initsizeof_fortran_t in sizeof_fortran_tmodule.o

      "_PyExc_RuntimeError", referenced from:

          _initsizeof_fortran_t in sizeof_fortran_tmodule.o

          _fortran_call in fortranobject.o

      "_PyExc_TypeError", referenced from:

          _fortran_call in fortranobject.o

          _array_from_pyobj in fortranobject.o

      "_PyExc_ValueError", referenced from:

          _array_from_pyobj in fortranobject.o

      "_PyImport_ImportModule", referenced from:

          _initsizeof_fortran_t in sizeof_fortran_tmodule.o

      "_PyMem_Free", referenced from:

          _fortran_dealloc in fortranobject.o

          _fortran_getattr in fortranobject.o

      "_PyMem_Malloc", referenced from:

          _fortran_getattr in fortranobject.o

      "_PyModule_GetDict", referenced from:

          _initsizeof_fortran_t in sizeof_fortran_tmodule.o

      "_PyOS_snprintf", referenced from:

          _fortran_getattr in fortranobject.o

      "_PyObject_GetAttrString", referenced from:

          _initsizeof_fortran_t in sizeof_fortran_tmodule.o

          _fortran_repr in fortranobject.o

      "_PyString_AsString", referenced from:

          _fortran_repr in fortranobject.o

      "_PyString_ConcatAndDel", referenced from:

          _fortran_getattr in fortranobject.o

      "_PyString_FromFormat", referenced from:

          _fortran_repr in fortranobject.o

      "_PyString_FromString", referenced from:

          _initsizeof_fortran_t in sizeof_fortran_tmodule.o

          _fortran_repr in fortranobject.o

          _fortran_getattr in fortranobject.o

      "_PyString_FromStringAndSize", referenced from:

          _fortran_getattr in fortranobject.o

      "_PyType_IsSubtype", referenced from:

          _array_from_pyobj in fortranobject.o

      "_PyType_Type", referenced from:

          _initsizeof_fortran_t in sizeof_fortran_tmodule.o

      "_Py_BuildValue", referenced from:

          _f2py_rout_sizeof_fortran_t_sizeof_fortran_t in sizeof_fortran_tmodule.o

      "_Py_FindMethod", referenced from:

          _fortran_getattr in fortranobject.o

      "_Py_InitModule4_64", referenced from:

          _initsizeof_fortran_t in sizeof_fortran_tmodule.o

      "__PyObject_New", referenced from:

          _PyFortranObject_New in fortranobject.o

          _PyFortranObject_NewAsAttr in fortranobject.o

      "__Py_NoneStruct", referenced from:

          _fortran_getattr in fortranobject.o

          _array_from_pyobj in fortranobject.o

          _fortran_setattr in fortranobject.o

      "_main", referenced from:

          start in crt1.10.6.o

    ld: symbol(s) not found for architecture x86_64

    collect2: error: ld returned 1 exit status

    error: Command "/Users/travis/miniconda/envs/_build/bin/gfortran -Wall -g -arch x86_64 build/temp.macosx-10.5-x86_64-2.7/build/src.macosx-10.5-x86_64-2.7/f90wrap/sizeof_fortran_tmodule.o build/temp.macosx-10.5-x86_64-2.7/build/src.macosx-10.5-x86_64-2.7/fortranobject.o build/temp.macosx-10.5-x86_64-2.7/f90wrap/sizeoffortran.o -L/Users/travis/miniconda/envs/_build/lib/gcc/x86_64-apple-darwin11.4.2/4.8.5 -L/Users/travis/miniconda/envs/_build/lib -lgfortran -o build/lib.macosx-10.5-x86_64-2.7/f90wrap/sizeof_fortran_t.so" failed with exit status 1
```

I'd hasten to guess that you are missing the python headers - do you need to configure the include path in some way?

**Comment by ocefpaf** [comment]
> There is more to the error:

I added the link to avoid polluting the tracker with the whole thing, but I guess we need it visible.

> I'd hasten to guess that you are missing the python headers - do you need to configure the include path in some way?

I have to put this on hold at the moment as this will be a little bit painful to fix/iterate without a OS X machine.

**Comment by jameskermode** [comment]
Hello, I'm the developer of the f90wrap package. I'd be happy to help with maintaining a conda recipe for it. I haven't understood what's going on here exactly - it is failing to build on an OS X machine, is that right? I use f90wrap on OS X with Python from conda so should be able to help. How do I test the recipe?

**Comment by ocefpaf** [comment]
> Hello, I'm the developer of the f90wrap package. I'd be happy to help with maintaining a conda recipe for it.

Thanks @jameskermode! I just listed you as a maintainer. That will give you commit rights to the `feedstock` (a separate repository for this recipe once it is merged here).

>  I haven't understood what's going on here exactly - it is failing to build on an OS X machine, is that right? I use f90wrap on OS X with Python from conda so should be able to help. How do I test the recipe?

Yes, the failure is on OS X only. BTW we are using the `gcc` available in conda's `default` channel. Which one are you using in your local config? You should be able to build this recipe by cloning/forking this repository and typing (from the recipes directory):

```shell
export CONDA_NPY=111  # We need to set a numpy version for the recipe.
conda build f90wrap
```

@pelson said in https://github.com/conda-forge/staged-recipes/pull/454#issuecomment-214715383 that we might be missing a way to specify the include directory. Not sure if we should patch the `setup.py` to search for that, or if there is an easy way to do it.

PS: We are building `f90wrap` based on a specific [commit](https://github.com/conda-forge/staged-recipes/pull/454/files#diff-01595e8989d78a3aefd17826da6a26ecR2). It would be nice to get an official release.

**Comment by jameskermode** [comment]
OK, sorry for the delay, I just got a change to start having a go at this now. I was previously using `gcc` from homebrew so the `conda build f90wrap` command triggered an install of gcc 4.8.5-3 and friends from conda.

When I compile `f90wrap` with ``python setup.py build` and my homebrew gcc, the final link line is:

```
/usr/local/bin/gfortran -Wall -g -m64 -Wall -g -undefined dynamic_lookup -bundle build/temp.macosx-10.5-x86_64-2.7/build/src.macosx-10.5-x86_64-2.7/f90wrap/sizeof_fortran_tmodule.o build/temp.macosx-10.5-x86_64-2.7/build/src.macosx-10.5-x86_64-2.7/fortranobject.o build/temp.macosx-10.5-x86_64-2.7/f90wrap/sizeoffortran.o -L/usr/local/Cellar/gcc/4.9.2_1/lib/gcc/4.9/gcc/x86_64-apple-darwin14.1.0/4.9.2 -L//anaconda/lib -lgfortran -o build/lib.macosx-10.5-x86_64-2.7/f90wrap/sizeof_fortran_t.so
```

while with `conda build f90wrap` it is

```
/anaconda/envs/_build/bin/gfortran -Wall -g -arch x86_64 build/temp.macosx-10.5-x86_64-2.7/build/src.macosx-10.5-x86_64-2.7/f90wrap/sizeof_fortran_tmodule.o build/temp.macosx-10.5-x86_64-2.7/build/src.macosx-10.5-x86_64-2.7/fortranobject.o build/temp.macosx-10.5-x86_64-2.7/f90wrap/sizeoffortran.o -L/anaconda/envs/_build/bin/../lib/gcc/x86_64-apple-darwin11.4.2/4.8.5 -L//anaconda/envs/_build/lib -lgfortran -o build/lib.macosx-10.5-x86_64-2.7/f90wrap/sizeof_fortran_t.so
```

So it's the ``-undefined dynamic_lookup -bundle build/temp.macosx`` which is missing from the conda build. Any ideas how to tell conda's gfortran to include this?

**Comment by ocefpaf** [comment]
@jakirkham do you have any idea how to do that on OS X? I am clueless here.

**Comment by jakirkham** [comment]
Yeah, not sure that I would recommend that route. Let me think about this a bit more.

**Comment by ocefpaf** [comment]
I am skipping `OS X` here as no one with a Mac was able to sort this out. Let's try to fix this in the `feedstock`.

File: recipes/f90wrap/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+requirements:
+    build:
+        - python
+        - setuptools
+        - numpy x.x
+        - gcc
+    run:
+        - python
+        - numpy x.x
+        - libgcc  # [linux]
```
Comment (jakirkham) [inline recipes/f90wrap/meta.yaml]: Might want to be careful with this.

**Pull request merged**